### PR TITLE
Write to file

### PR DESCRIPTION
**Pull request by graceolivia**
Added the ability to write to a file! It will automatically go in the new `Charts` folder and create a .txt file. Also added a mock to ensure the file is printed.

### CODE REVIEW

File: Sources/SwiftCli/main.swift [hunk 1]
```diff
@@ -12,14 +12,18 @@ struct StartProgram: ParsableCommand{
...
 
+    @Option(help: "File name to save pattern under.")
+    var outputFile: String?
+
 
     func run() {
         let inputValidator = InputValidator()
         let chartConstructor = ChartConstructor()
         let fileValidator = FileValidator()
+        let writeToFile = WriteToFile()
```
Comment (breadoliveoilsalt) [inline Sources/SwiftCli/main.swift:23]: Perhaps a name change to `FileWriter`?

File: Tests/SwiftCliCoreTests/ChartifyTest.swift [hunk 2]
```diff
@@ -7,23 +7,35 @@ class ChartifyFinishedTest: XCTestCase {
...
+        Chartify(inputValidator: InputValidator(), chartConstructor: ChartConstructor(), fileValidator: FileValidator(), writeToFile: mock).run(userInput: ["k1"], fileNameToWrite: Optional("test"))
+        expect(mock.wasWriteToFileCalled).to(equal(true))
+    }
+
+    func testRunDoesntCallMockWriteToFileIfUserDoesntIncludeWriteToFile() throws {
+        let mock = MockWriteToFile()
+        Chartify(inputValidator: InputValidator(), chartConstructor: ChartConstructor(), fileValidator: FileValidator(), writeToFile: mock).run(userInput: ["k1"])
+        expect(mock.wasWriteToFileCalled).to(equal(false))
+    }
+
```
Comment (breadoliveoilsalt) [inline Tests/SwiftCliCoreTests/ChartifyTest.swift:38]: I like where these are.  Nice job.  Small nit: no need to include `Mock` in the description of the test (function name).

File: Sources/SwiftCliCore/WriteToFile.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class WriteToFile {
+    public init() {}
+
+    public func writeFile(chart: String,
+                          filePath: String,
+                          fileName: String) throws {
+
+        do {
+            try chart.write(toFile: filePath + "//" + fileName + ".txt",
+                            atomically: true,
```
Comment (breadoliveoilsalt) [inline Sources/SwiftCliCore/WriteToFile.swift:12]: Looks like a wrapper 👍

File: Sources/SwiftCliCore/Chartify.swift [hunk 4]
```diff
@@ -6,17 +6,19 @@ public final class Chartify {
...
 
         self.inputValidator = inputValidator
         self.chartConstructor = chartConstructor
         self.fileValidator = fileValidator
+        self.writeToFile = writeToFile
     }
 
 
-    public func run(userInput: [String], file: String? = nil, knitFlat: Bool = false) {
+    public func run(userInput: [String], file: String? = nil, knitFlat: Bool = false, fileNameToWrite: String? = nil) {
```
Comment (hnlee) [inline Sources/SwiftCliCore/Chartify.swift:21]: Also nonblocking for this PR but: does `Chartify` need to know or care about what output file name to write to? Or is that something that it makes more sense for `FileWriter` to know?

File: Sources/SwiftCliCore/Chartify.swift [hunk 5]
```diff
@@ -30,7 +32,17 @@ public final class Chartify {
...
+
+            if let fileNameToWrite = fileNameToWrite {
+                do {
+                    try writeToFile.writeFile(chart: chart, filePath: "Charts", fileName: fileNameToWrite)
+                } catch {
+                    print(error.localizedDescription)
+                    exit(0)
+                }
+            } else {
+                print(chart) }
```
Comment (hnlee) [inline Sources/SwiftCliCore/Chartify.swift:44]: Looks good but maybe worth extracting into its own private method.

**Review by hnlee — CHANGES_REQUESTED** [review]

File: Sources/SwiftCliCore/Chartify.swift (hunk 5 above)
Comment (graceolivia) [reply Sources/SwiftCliCore/Chartify.swift]: Done!

File: Tests/SwiftCliCoreTests/ChartifyTest.swift [hunk 6]
```diff
@@ -58,3 +70,15 @@ class MockFileValidator: FileValidator {
         return ["k1"]
     }
 }
+
+class MockWriteToFile: FileWriter {
```
Comment (hnlee) [inline Tests/SwiftCliCoreTests/ChartifyTest.swift:74]: Calling it `MockFIleWriter` makes it clearer what you're mocking.

File: Tests/SwiftCliCoreTests/ChartifyTest.swift [hunk 7]
```diff
@@ -58,3 +70,15 @@ class MockFileValidator: FileValidator {
...
+    var wasWriteToFileCalled = false
+
+    override func writeFile(chart: String,
+                            filePath: String,
+                            fileName: String) {
+
+        wasWriteToFileCalled = true
```
Comment (hnlee) [inline Tests/SwiftCliCoreTests/ChartifyTest.swift:81]: Having naming consistency, e.g. `wasWriteFileCalled`, makes it clearer which method you're checking.

File: Sources/SwiftCliCore/Chartify.swift [hunk 8]
```diff
@@ -44,4 +47,18 @@ public final class Chartify {
         return (chartConstructor.makeChart(patternMetaData: metaData))
 
     }
+
+    private func returnChartInPrefferedFormat(chart: String, fileNameToWrite: String?) throws {
```
Comment (hnlee) [inline Sources/SwiftCliCore/Chartify.swift:51]: Typo in the name here

**Review by hnlee — CHANGES_REQUESTED** [review]
Some naming changes requested, but after that it should be good to merge.

**Pull request merged**